### ISSUE TITLE
[Fatal] Array and string offset access syntax with curly braces is no longer supported | wc-formatting-functions.php:791

### ISSUE THREAD

**Issue by nigeljamesstevenson** [woothemes/woocommerce#53734]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

link to code

```
PHP Fatal error:  Array and string offset access syntax with curly braces is no longer supported in /../wp-content/plugins/woocommerce/includes/wc-formatting-functions.php on line 791
``` 

### Expected behavior

No fatal error

### Actual behavior

Fatal error witnessed

### Steps to reproduce

We don’t have steps to reproduce, but 742 Fatals of this type were observed on Fri Dec 6, 2024 across 11 sites.
Hopefully, someone can dig in and apply a defensive check if appropriate. – To be updated after investigation

### WordPress Environment

N/A

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by nigeljamesstevenson** [comment]
This one appears similar to [[Fatal] Array and string offset access syntax with curly braces is no longer supported | wc-formatting-functions.php](https://github.com/woocommerce/woocommerce/issues/51882#top)

**Comment by barryhughes** [comment]
Hey @nigeljamesstevenson! Though there is a different line number this time round, I would still categorize this as being essentially the same issue as before. It happens if a relatively modern PHP runtime encounters code like this:

```php
$array_or_string{100};
```

I don't see any traces of this in the [cited file](https://github.com/woocommerce/woocommerce/blob/9.5.1/plugins/woocommerce/includes/wc-formatting-functions.php), which is why we closed the earlier bug report. Also, just for additional clarity, I'll highlight that you *may* still see code like the following in the same file, and across the codebase, which is perfectly valid:

```php
"String {$interpolated_var}";
```

Definitely correct me if I'm mistaken, but the environments from which we collect these errors generally let us infer the WooCommerce version from the filepath (something like `/var/.../woocommerce/9.5.0/source.php`) whereas, for these particular errors, we don't have that data and don't therefore know which version of WooCommerce is in play. Is that correct? If so, I think that further underlines that the problem is either in a much older version of WooCommerce, or else a copy of WooCommerce that has been modified in some way.

I'd recommend we close this, but happy to leave it open if we want to wait and try to collect more information.

**Comment by albarin** [comment]
Hey @nigeljamesstevenson, I've looked further into this and I think Barry is right here.
I've searched the history of the file and I've found the PR where that particular issue was fixed (exact same line as the error): https://github.com/woocommerce/woocommerce/pull/24460/files, they also mention in comments they think this is the [last occurrence](https://github.com/woocommerce/woocommerce/pull/24460#issuecomment-524045004) of these curly braces deprecation fixes.

So, I think we cannot do much here, the fatal is coming from a site with a very old version of WC (the fix is from 2019), should we close this, wdyt?

(cc: @jorgeatorres)

**Comment by jorgeatorres** [comment]
I agree @albarin. Also, a quick search across the database (9.6) didn't reveal any other uses of the deprecated syntax, which seems to match what those discussing it in the original PR wrote. I believe we should be good to close.

**Comment by nigeljamesstevenson** [comment]
Thanks all 🙌
